code/api/lecture_search/engine.py

```diff
@@ -47,6 +48,8 @@ def query_corpus(self, query_txt, max_results):
             video_id = metadata.get("video_id")
             file_identifier = metadata.get("AWS_file")
             aws_path = "https://cs410videostorage.s3.amazonaws.com/"
+            txt_path = "{0}{1}.txt".format(aws_path, file_identifier)
```

Comment:
Would it better to get the file contents from the file system under the corpora directory, rather than make a call to AWS for each file?

Something like this:

```
text_path = "corpora/lectures/{}.txt".format(video_id)
with open(text_path) as file:
    full_text = file.read();

search_result = {
    "03_video_id": video_id,
    "01_doc_id": doc_id,
    "02_score": float("{:.3f}".format(score)),
    "00_rank": i + 1,
    "04_title": metadata.get("title"),
    "05_vid_path": "{0}{1}.mp4".format(aws_path, file_identifier),
    "06_txt_path": "{0}{1}.txt".format(aws_path, file_identifier),
    "07_pdf_path": "{0}{1}.pdf".format(aws_path, file_identifier),
    "08_full_text": full_text
}
```

Reply:
Absolutely agree. That was actually my first instinct but I didn't want to make any assumptions about how the file structure works when deployed to the client (didn't realize it's precisely as defined in the repo).

Reply:
Oh yes, the file structure on the server is the exact same as the repository. I can push a commit to this branch to read file from the file system if you'd like.

Reply:
Works for me...thank you, Rick!

Reply:
Done!